docs/dev-proc/smbios-rules.md

```diff
@@ -0,0 +1,98 @@
+# DMI/SMBIOS filling convention
+
+2 types of devices may be supported by Dasharo:
+
+- Those that already have a supported firmware implementation available
+- Those that do not
+
+In the first case, a device may have drivers available in Windows Update or
+firmware via fwupd/LVFS. If that is the case, we want to leverage this support
+in Dasharo too. Since fwupd and Windows Update identify devices by (among other
+things) SMBIOS contents, we do this by matching as many SMBIOS fields as
+possible.
+
+For example, in fwupd, the following SMBIOS fields are used for generating
+device GUIDs:
+
+```bash
+Computer Information
+--------------------
+BiosVendor: 3mdeb Embedded Systems Consulting
+BiosVersion: Dasharo (coreboot+UEFI) v1.0.0
+BiosMajorRelease: 1
+BiosMinorRelease: 0
+Manufacturer: Notebook
+Family: Not Applicable
+ProductName: NV4XMB,ME,MZ
+ProductSku: Not Applicable
+EnclosureKind: a
+BaseboardManufacturer: Notebook
+BaseboardProduct: NV4XMB,ME,MZ
+
+Hardware IDs
+------------
+{37b192b9-4fe8-5d20-b4e7-6d00782e2a09}   <- Manufacturer + Family + ProductName + ProductSku + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
+{bc90b4cf-a507-50c5-bf51-48c26863dc13}   <- Manufacturer + Family + ProductName + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
+{cf004452-c518-5899-a791-d246db94c413}   <- Manufacturer + ProductName + BiosVendor + BiosVersion + BiosMajorRelease + BiosMinorRelease
+{25b6ea34-8f52-598e-a27a-31e03014dbe3}   <- Manufacturer + Family + ProductName + ProductSku + BaseboardManufacturer + BaseboardProduct
+{dd71301d-2915-5884-af96-498003ea55bb}   <- Manufacturer + Family + ProductName + ProductSku
+{8603f686-41b8-5ea0-bbf8-d3e5cfa9c0ac}   <- Manufacturer + Family + ProductName
+{17e12139-ddb8-5e43-ad90-74e69a044051}   <- Manufacturer + ProductSku + BaseboardManufacturer + BaseboardProduct
+{cd9d0da1-e88d-5d36-aae5-2ba60e8d0ed6}   <- Manufacturer + ProductSku
+{99192755-edd8-5f6d-8c46-e7d9e0b8be5f}   <- Manufacturer + ProductName + BaseboardManufacturer + BaseboardProduct
+{0f4d8e7f-9419-5817-a8c5-aa54027c1b35}   <- Manufacturer + ProductName
+{17e12139-ddb8-5e43-ad90-74e69a044051}   <- Manufacturer + Family + BaseboardManufacturer + BaseboardProduct
+{cd9d0da1-e88d-5d36-aae5-2ba60e8d0ed6}   <- Manufacturer + Family
+{a127c2d7-5537-5b72-8bc1-9ccd683f18e6}   <- Manufacturer + EnclosureKind
+{96005f57-a38f-5b5e-a26a-50d7286811b0}   <- Manufacturer + BaseboardManufacturer + BaseboardProduct
+{56bb94fc-caf1-5863-9e85-1e449a0c5a01}   <- Manufacturer
+```
+
+In this case, we set all the fields that are relevant (excluding BIOS version
+and vendor) to the same values as the stock firmware.
```

Comment:
At the end of document I realized this should be developers documentation, since it is in development section of the website, but as I wrote above then it should be more step by step guide/process of getting correct values for given fields that have to filled. Maybe even section dedicated to coreboot-specific procedures.

This document is more like end-user explanation then developer documentation. Please be clear at the beginning what audience this document is for.

Reply:
Added an introduction describing what this document is - guidelines for developers. Restructured the document accordingly